File: csharp/bob/Bob.cs
using System;
using System.Linq;

public static class Bob
{
    public static string Response(string statement)
    {
        if (string.IsNullOrWhiteSpace(statement)) return "Fine. Be that way!";

        statement = statement.Trim();
        if (statement.Last() == '?')
        {
            if (statement == statement.ToUpper() && statement.Any(char.IsLetter))
            {
                return "Calm down, I know what I'm doing!";
            }
            else
            {
                return "Sure.";
            }
        }
        else if (statement == statement.ToUpper() && statement.Any(char.IsLetter))
        {
            return "Whoa, chill out!";
        }
        else
        {
            return "Whatever.";
        }
    }
}

File: fsharp/bob/Bob.fs
module Bob

let (|Shout|_|) (input: string) =
    if input = input.ToUpper() && input <> input.ToLower() then Some Shout else None

let (|Question|_|) (input: string) =
    if input.EndsWith('?') then Some Question else None

let response (input: string): string =
    match input.Trim() with 
    | "" -> "Fine. Be that way!"
    | Question & Shout -> "Calm down, I know what I'm doing!"
    | Question -> "Sure."
    | Shout -> "Whoa, chill out!"
    | _ -> "Whatever."